mbed_greentea/mbed_test_api.py

```diff
@@ -335,6 +357,38 @@ def get_auto_property_value(property_name, line):
     end_time = time()
     testcase_duration = end_time - start_time   # Test case duration from reset to {end}
 
+    def pack_hex_payload(path, payload):
+        """! Convert a block of hex string data back to binary and return the binary data
+        @param path Where to save file with binary payload
+        @param payload String with hex encoded ascii data, e.g.: '6164636772...'
+        @return bytearray with payload with data
+        """
+        hex_pairs = map(''.join, zip(*[iter(payload)] * 2)) # ['61', '64', '63', '67', '72', ... ]
+        bin_payload = bytearray([int(s, 16) for s in hex_pairs])
+        return bin_payload
+
+    def pack_base64_payload(path, payload):
+        """! Convert a block of base64 data back to binary and return the binary data
+        """
+        return None
+
+    # We may have code coverage data to dump from test prints
+    if coverage_start_data_list:
+        gt_log("storing coverage data artefacts")
+        for cov_data in coverage_start_data_list:
+            path = os.path.abspath(cov_data['path'])
+            payload = cov_data['payload']
+            encoding = cov_data['encoding']
+            bin_payload = None
+            if encoding == 'hex':
```

Comment:
We do not support for now, added this in case Brendan wants to use Base64. Now it looks like we will not use it.
If not I will clean this PR from base64 encoding code :)

@0xc0170 This is not production level PR, this code on this branch is used to check if Brendan's solution works for us. If yes, I will make this PR production ready.